src/utils/SocketClient.hxx

```diff
@@ -59,6 +60,7 @@ public:
         MDNS_CONNECT,
         STATIC_CONNECT,
         CONNECT_FAILED,
+        CONNECT_FAILED_SELF,
```

Comment:
i'm not sure we need to have a separate status returned for this. It will make the UI more complex. We could report the same as CONNECT_FAILED.

Reply:
This was added primarily for unit test purposes.